src/kafka_cffi/message.py

```diff
@@ -24,6 +25,25 @@ def __init__(self, rkmessage):
         tstype = ffi.new("rd_kafka_timestamp_type_t *")
         ts = lib.rd_kafka_message_timestamp(rkmessage, tstype)
         self.__timestamp = (tstype[0], ts)
+
+        c_header_pt = ffi.new("rd_kafka_headers_t **")
+        lib.rd_kafka_message_headers(rkmessage, c_header_pt)
+        c_header = c_header_pt[0]
+        hdcount = lib.rd_kafka_header_cnt(c_header)
+
+        c_hd_key = ffi.new("const char **")
+        c_hd_payload = ffi.new("const char **")
+        c_hd_size = ffi.new("size_t *")
+        self.__headers = []
+        for idx in range(hdcount):
+            error = lib.rd_kafka_header_get_all(
+                c_header, idx, c_hd_key, ffi.cast("const void **", c_hd_payload), c_hd_size)
+            if error == 0:
+                self.__headers.append(
+                    (ffi.string(c_hd_key[0]),
+                     ffi.string(c_hd_payload[0], int(c_hd_size[0])))
```

Comment:
This is potentially binary data, right? Should we be using `ffi.unpack` instead of `ffi.string` here?

Reply:
You are right. I need to do more checking on the type of header value before casting I guess

Reply:
If I'm understanding this correctly, the header value is binary data, so perhaps we should just leave it as binary data (`bytes` in python3, as returned by `ffi.unpack`), and let the API user handle it?

Reply:
@hyperair 💇‍♂